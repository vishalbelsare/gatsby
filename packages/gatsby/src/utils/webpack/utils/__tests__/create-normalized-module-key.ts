import path from "path"
import { createNormalizedModuleKey } from "../create-normalized-module-key"

const projectRoot = path.join(`Users`, `username`, `project`)

it(`should normalize bare module imports from node_modules`, () => {
  const nodeModule = path.join(`node_modules`, `package-name`)

  const normalizedModuleKey = createNormalizedModuleKey(
    path.join(projectRoot, nodeModule),
    projectRoot
  )

  expect(normalizedModuleKey).toMatchSnapshot()
})

it(`should normalize relative module imports from node_modules`, () => {
  const nodeModule = path.join(`node_modules`, `package-name`)

  const normalizedModuleKey = createNormalizedModuleKey(
    path.join(projectRoot, nodeModule, `index.js`),
    projectRoot
  )

  expect(normalizedModuleKey).toMatchSnapshot()
})

it(`should normalize local module imports`, () => {
  const rootRelativeLocalModuleFile = path.join(`src`, `components`, `index.js`)
  const absoluteLocalModuleFile = path.join(
    projectRoot,
    rootRelativeLocalModuleFile
  )

  const normalizedModuleKey = createNormalizedModuleKey(
    absoluteLocalModuleFile,
    projectRoot
  )
  expect(normalizedModuleKey).toMatchSnapshot()
})
