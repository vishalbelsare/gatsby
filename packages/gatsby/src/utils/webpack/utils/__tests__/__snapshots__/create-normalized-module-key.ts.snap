// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should normalize bare module imports from node_modules 1`] = `"file://node_modules/package-name"`;

exports[`should normalize local module imports 1`] = `"file://src/components/index.js"`;

exports[`should normalize relative module imports from node_modules 1`] = `"file://node_modules/package-name"`;
